src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -678,6 +678,18 @@ private void EmitReturnStatement(BoundReturnStatement boundReturnStatement)
                 }
                 else
                 {
+                    if (expressionOpt != null)
+                    {
+                        // Ensure the return type has been translated. (Necessary
+                        // for cases of untranslated anonymous types.)
+                        var returnType = expressionOpt.Type;
+                        var byRefType = returnType as ByRefReturnErrorTypeSymbol;
+                        if ((object)byRefType != null)
+                        {
```

Comment:
There are tests that hit this code path with ByRef return types, although we do not generate methods with ByRef anonymous types.